tests/Resources/ti.geolocation.test.js

```diff
@@ -307,12 +304,135 @@ describe.windowsBroken('Titanium.Geolocation', () => {
+					if (!e.success) {
+						return finish(new Error('Failed to get Geolocation ALWAYS permission'));
+					}
+					const result = Ti.Geolocation.getCurrentHeading();
+					result.should.be.a.Promise();
+					if (Ti.Geolocation.hasCompass) {
+						result.then(data => {
+							should(data).have.property('success').which.is.true();
+							should(data).have.property('code').which.eql(0);
+							should(data.heading).be.an.Object();
+							should(data.heading).have.a.property('magneticHeading').which.is.a.Number();
+							should(data.heading).have.a.property('accuracy').which.is.a.Number();
+							return finish();
+						}).catch(e => finish(e));
+					} else {
+						// expect to fail if no compass!
+						result.then(() => finish(new Error('Expected to fail getCurrentHeading() with no compass support!'))).catch(e => finish());
+					}
+				});
+			});
 		});
 
 		describe('#getCurrentPosition()', () => {
 			it('is a Function', () => {
 				should(Ti.Geolocation).have.a.property('getCurrentPosition').which.is.a.Function();
 			});
+
+			// FIXME: On local emulator these fail because passive/network providers are not available
+			it('works via callback argument', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				Ti.Geolocation.requestLocationPermissions(Ti.Geolocation.AUTHORIZATION_ALWAYS, function (e) {
+					if (!e.success) {
+						return finish(new Error('Failed to get Geolocation ALWAYS permission'));
+					}
+
+					Ti.Geolocation.getCurrentPosition(function (data) {
+						try {
+							should(data).have.property('success').which.is.true();
+							should(data).have.property('code').which.eql(0);
+							should(data.coords).be.an.Object();
+							should(data.coords).have.a.property('latitude').which.is.a.Number();
+							should(data.coords).have.a.property('longitude').which.is.a.Number();
+							should(data.coords).have.a.property('altitude').which.is.a.Number();
+							should(data.coords).have.a.property('accuracy').which.is.a.Number();
+							should(data.coords).have.a.property('heading').which.is.a.Number();
+							should(data.coords).have.a.property('speed').which.is.a.Number();
+							should(data.coords).have.a.property('timestamp').which.is.a.Number();
+						} catch (err) {
+							console.log(data);
+							return finish(err);
+						}
+						finish();
+					});
+				});
+			});
+
+			it('works via Promise return value', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				Ti.Geolocation.requestLocationPermissions(Ti.Geolocation.AUTHORIZATION_ALWAYS, function (e) {
+					if (!e.success) {
+						return finish(new Error('Failed to get Geolocation ALWAYS permission'));
+					}
+
+					const result = Ti.Geolocation.getCurrentPosition(event => console.log(event));
+					try {
+						result.should.be.a.Promise();
+					} catch (err) {
+						return finish(err);
+					}
+					result.then(data => {
+						try {
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.geolocation.test.js;
  Line: 414;
-->

- :warning: tests/Resources/ti.geolocation.test.js line 414 – Each then() should return a value or throw (promise/always-return)